### PR TITLE
[integration] Refactoring AdvancedPricingImportExport fixture

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This PR refactores AdvancedPricingImportExport fixture.

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php [hunk 1]
```diff
@@ -4,26 +4,45 @@
...
-    ->setIsObjectNew(true)
-    ->save();
+    ->setIsObjectNew(true);
+
+$productRepository->save($productModel);
 
 $productModel->setName('AdvancedPricingSimple 2')
     ->setId(null)
-    ->setUrlKey(null)
+    ->setUrlKey('')
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php]: Why you changed type null to empty string here?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php (hunk 1 above)
Comment (Usik2203) [reply dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php]: Hi @ihor-sviziev  
I see that we shoud set `string` for `setUrlKey`
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Model/Product.php#L32
And We will get error if we use `declare(strict_types=1);` and set `null` to `setUrlKey`
Thanks

Comment (ihor-sviziev) [reply dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php]: @Usik2203 theoretically it's true, but practically you've changed type of argument, that might lead to generating incorrect url key or not generating url key at all.
I think it should be better to create separate object and not reuse $productModel there, so you'll not need to run setUrlKey at all there.

What do you thing?

**Comment by Usik2203** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7603](https://jira.corp.magento.com/browse/ENGCOM-7603) has been created to process this Pull Request

**Review by rogyar — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7603](https://jira.corp.magento.com/browse/ENGCOM-7603) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Charlie** [comment]
Failed tests not related to changes from this PR.

**Pull request merged**